pyigm/cgm/utils.py

```diff
@@ -258,7 +259,13 @@ def cgm_from_galaxy_igmsystems(galaxy, igmsystems, rho_max=300*u.kpc, dv_max=400
         # Loop to generate
         cgm_list = []
         for imatch in match:
-            cgm = CGMAbsSys(galaxy, igmsystems[imatch], cosmo=cosmo, **kwargs)
+            # Instantiate new IGMSystem
+            # Otherwise, updates to the IGMSystem cross-pollinate other CGMs
+            sysmatch = igmsystems[imatch]
+            newisys = IGMSystem(sysmatch.coord,sysmatch.zabs,
```

Comment:
have been advised to avoid copy statements when we can.
Can we?

Reply:
Hmm...part of the issue was that updates were propagating through the AbsComponent objects too!  In fact, merely copying the IGMSystem didn't fix the problem, so the deepcopy was the best solution I could think of.  We could of course reinstantiate AbsComponents, but that's less than ideal for obvious reasons...suggestions?

Reply:
I probably need to better understand the problem here.

If one does update the IGMSystem, wouldn't you want that
to propagate?  If not, I do feel you should re-instantiate.

Reply:
Just to clarify, yes, one would want the main attributes of a component (column density, b-value, transitions) to propagate but the same IGM system will have different 'systemic' component velocities when associated to one galaxy (CGM systems) versus another.  The .copy methods are very useful anyway, so I'll go that direction.